### PR TITLE
CVSL-1127 tidying up updating additional conditions code

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt [hunk 1]
```diff
@@ -117,47 +117,27 @@ class LicenceConditionService(
...
+    val removedConditions = existingConditions.getRemovedConditions(submittedConditions, request)
 
-    submittedAdditionalConditions.forEach { newAdditionalCondition ->
-      updatedAdditionalConditions.filter {
-        it.conditionCode == newAdditionalCondition.conditionCode
-      }.forEach {
-        it.conditionCategory = newAdditionalCondition.conditionCategory
-        it.conditionText = newAdditionalCondition.conditionText
-        it.conditionSequence = newAdditionalCondition.conditionSequence
-        it.conditionType = newAdditionalCondition.conditionType
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt:147]: not sure why this doesn't update the expanded text too!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt [hunk 2]
```diff
@@ -117,47 +117,27 @@ class LicenceConditionService(
...
-    val submittedAdditionalConditions =
+    val submittedConditions =
       request.additionalConditions.transformToEntityAdditional(licenceEntity, request.conditionType)
 
-    val newAdditionalConditions =
-      submittedAdditionalConditions.filter { licenceEntity.additionalConditions.none { submittedCondition -> submittedCondition.conditionCode == it.conditionCode } }
-        .map { newAdditionalCondition ->
-          newAdditionalCondition.copy(
-            expandedConditionText = newAdditionalCondition.conditionText,
-            licence = licenceEntity,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt:128]: don't need to set the licence as that's already done in the transform

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt [hunk 3]
```diff
@@ -117,47 +117,27 @@ class LicenceConditionService(
...
-          )
-        }
+    val existingConditions = licenceEntity.additionalConditions
 
-    val removedAdditionalConditionsList =
-      licenceEntity.additionalConditions.filter {
-        it.conditionType == request.conditionType && submittedAdditionalConditions.none { submittedCondition -> submittedCondition.conditionCode == it.conditionCode }
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt:134]: not sure why this checks on conditionType whereas the other ones don't. 
I think we have unique codes across AP and PSS so this probably isn't necessary (I've kept it in)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt (hunk 1 above)
Comment (gurps-singh) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt:147]: I think this updating for expanded text is done in the `updateAdditionalConditionData` function in the same service. No idea why?

**Review by gurps-singh — APPROVED** [review]
Very useful for the auditing ticket I am working on so thanks!

**Pull request merged**